components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java

```diff
@@ -0,0 +1,288 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.analytics.permissions.internal.dao;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.analytics.permissions.bean.Permission;
+import org.wso2.carbon.analytics.permissions.bean.PermissionConfig;
+import org.wso2.carbon.analytics.permissions.bean.Role;
+import org.wso2.carbon.analytics.permissions.exceptions.PermissionException;
+import org.wso2.carbon.datasource.core.api.DataSourceService;
+import org.wso2.carbon.datasource.core.exception.DataSourceException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+/**
+ * Permission DAO.
+ */
+public class PermissionsDAO {
+    private static final Logger log = LoggerFactory.getLogger(PermissionsDAO.class);
+
+    private DataSourceService dataSourceService;
+    private DataSource dataSource;
+    private PermissionConfig permissionConfig;
+
+    public PermissionsDAO(DataSourceService dataSourceService, PermissionConfig permissionConfig) {
+        this.dataSourceService = dataSourceService;
+        this.permissionConfig = permissionConfig;
+    }
+
+    /**
+     * Add permission.
+     *
+     * @param permission
+     */
```

Comment:
We have decided to add queries as the debug logs right? can we do that?

Reply:
Already added the queries to the debug logs below.

```java
...
ps.setString(2, permission.getPermissionString());
log.debug("Executing query: " + query);
ps.executeUpdate();
...
```